app/src/main/shared/Helpers.ts

```diff
@@ -36,5 +39,24 @@ export const helpers = {
             obj[key] = value;
         });
         return obj;
+    },
+    ingestQueryStringAndReturnResult<TModel>(): Result | void {
+
+        const queryAsObject = this.queryStringAsObject();
+
+        if (!queryAsObject.result) {
+            return null
+        }
+
+        try {
+            const decoded = jwtDecoder.jwtDecode(queryAsObject.result);
+            history.replaceState({}, document.title, location.href.split("?")[0]);
+            return JSON.parse(decoded.result);
+        }
+        catch (e) {
+            // if the query string token is nonsense, just return null
+            history.replaceState({}, document.title, location.href.split("?")[0]);
```

Comment:
No need to repeat this statement - separate it out into a helpfully named function?